### ISSUE TITLE
empty labels for checkbox items are removed

### ISSUE THREAD

**Issue by DavidBruchmann** [sabbelasichon/typo3-rector#4238]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | at least 2.2 - 2.4           |

In definitions like this:
```
            'config' => [
                'type' => 'check',
                'items' => [
                    [
                        'label' => '',
                    ],
                ],
            ],
```
the item-array gets completely removed by the typo3-rector rules `SimplifyCheckboxItemsTCARector` and `MigrateItemsIndexedKeysToAssociativeRector`, both have to be excluded to keep the definitions.

## Minimal PHP Code Causing Issue

Sorry, I don't know how to load the Ssch namespace there:
https://getrector.com/demo/890354e6-9ea1-439a-95d7-fe7bf1e798eb

## Expected Behaviour

the "items" array is required by TYPO3 and shouldn't be removed entirely.
One item with label is also required, even if it's empty.

**Comment by DavidBruchmann** [comment]
[Docu v12](https://docs.typo3.org/m/typo3/reference-tca/12.4/en-us/ColumnsConfig/Type/Check/Default.html)  
[Docu v11](https://docs.typo3.org/m/typo3/reference-tca/11.5/en-us/ColumnsConfig/Type/Check/Default.html)

**Comment by simonschaufi** [comment]
Changelogs: 
- https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.3/Deprecation-99739-IndexedArrayKeysForTCAItems.html
- https://review.typo3.org/c/Packages/TYPO3.CMS/+/72056

Can you please have a a look at both?

I guess this one causes the problem: https://review.typo3.org/c/Packages/TYPO3.CMS/+/72056

**Comment by DavidBruchmann** [comment]
Yes I agree, I just don't know how to make TYPO3 and rector happy with the same notation.
Perhaps @brotkrueml can shine a light on it?

**Comment by DavidBruchmann** [comment]
your example looks outdated as it never has the `label` key

**Comment by DavidBruchmann** [comment]
ok, I can't follow completely, as my example looks differently, but let's see what you're coming forward with
